### ISSUE TITLE
Warm start with pretrained models

### ISSUE THREAD

**Issue by expectopatronum** [zhiCHEN96/ConceptWhitening#4]
Hi everyone!
Thanks for the great paper and publishing the code! I am currently trying to implement concept whitening for one of my own datasets and have a few questions regarding "Warm start with pretrained models".

1. "The model usually converges in one epoch (one pass over the data) if a pretrained model is used." Does this refer to one pass over the auxiliary data or are you performing one pass over the auxiliary data and another pass over the training data for the main task/objective?
2. "In order to limit the total time of the training process, we used pretrained models for the popular CNN architectures (discussed above) and fine-tuned these models after BN was replaced with CW." / "if a pretrained model (using BN) exists, one can simply replace the BN module with a CW module and train it for one epoch, in which case, the pretrained black-box model can be turned into a more interpretable model that is approximately equally accurate." same here, does "replace the BN module with a CW module" already include aligning the concepts? Or does "train it for one epoch" refer to aligning the concepts using the auxiliary data?

I am looking forward to you clarification and apologize in case I missed something obvious in the paper.
Thanks and best regards
Verena

**Comment by zhiCHEN96** [comment]
Thanks for being interested in our work!

The "one epoch" here refers to one epoch of the main dataset (e.g. Places365) rather than the auxiliary concept dataset.

In our code, a two step alternating optimization is used for the two objectives, the main classification objective and the concept alignment objective. If you take a look at line 268 in train_place.py, you can find that we optimize the concept alignment matrix every 30 iterations of training for the main objective. So the concept alignment is trained together with the main objective, i.e. repeating of 30 iterations of main task and 1 iteration of concept alignment, and the "one epoch" refers to one pass of the main dataset.

**Comment by expectopatronum** [comment]
Thanks for the clarification!

Ok, I noticed the alternating optimization in the paper, but how I understood the section "Warm start with pretrained models" and the description in section 4.1 is that we either 1) use the alternating approach when training a model from scratch or 2) use a pretrained model, replace one layer and align the concepts. (before you clarified it in your answer)
But I think I got it now!

One more question: In your experiments, did you ever align the concepts throughout the whole training process or is it always a pretrained model + concept aligning in one additional epoch?

Thanks a lot & best regards
Verena
